test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -1041,13 +1077,54 @@ public static void forceFail(String reason) {
         latch.countDown();
     }
 
+    /**
+     * Adds a {@code message} to the log area, if enabled by
+     * {@link Builder#logArea()} or {@link Builder#logArea(int)}.
+     *
+     * @param message to log
+     */
+    public static void log(String message) {
+        invokeOnEDTUncheckedException(() -> {
+            if (logArea != null) {
+                logArea.append(message + System.lineSeparator());
+            }
```

Comment:
```suggestion
            if (logArea != null) {
                logArea.append(message + "\n");
            }
```
Text components in Swing use `\n` as line separator.